src/cdk/overlay/overlay-ref.ts

```diff
@@ -127,73 +135,80 @@ export class OverlayRef implements PortalHost {
     this._detachments.complete();
   }
 
-  /**
-   * Checks whether the overlay has been attached.
-   */
+  /** Whether the overlay has attached content. */
   hasAttached(): boolean {
     return this._portalHost.hasAttached();
   }
 
-  /**
-   * Returns an observable that emits when the backdrop has been clicked.
-   */
+  /** Gets an observable that emits when the backdrop has been clicked. */
   backdropClick(): Observable<void> {
     return this._backdropClick.asObservable();
   }
 
-  /** Returns an observable that emits when the overlay has been attached. */
+  /** Gets an observable that emits when the overlay has been attached. */
   attachments(): Observable<void> {
     return this._attachments.asObservable();
   }
 
-  /** Returns an observable that emits when the overlay has been detached. */
+  /** Gets an observable that emits when the overlay has been detached. */
   detachments(): Observable<void> {
     return this._detachments.asObservable();
   }
 
-  /**
-   * Gets the current state config of the overlay.
-   */
-  getState(): OverlayConfig {
-    return this._state;
+  /** Gets the the current overlay configuration, which is immutable. */
+  getConfig(): ImmutableObject<OverlayConfig> {
+    // Clone the config so that it cannot be modified outside of this class.
+    return this._config;
   }
 
   /** Updates the position of the overlay based on the position strategy. */
   updatePosition() {
-    if (this._state.positionStrategy) {
-      this._state.positionStrategy.apply();
+    if (this._config.positionStrategy) {
+      this._config.positionStrategy.apply();
     }
   }
 
+  /** Update the size properties of the overlay. */
+  updateSize(sizeConfig: OverlaySizeConfig) {
+    this._config = {...this._config, ...sizeConfig};
+    this._updateElementSize();
+  }
+
+  /** Sets the LTR/RTL direction for the overlay. */
+  setDirection(dir: Direction) {
+    this._config = {...this._config, direction: dir};
```

Comment:
Why clone the config here instead of just setting the direction?

Reply:
Because the config is immutable

Reply:
Shouldn't it be enough to clone it on init and then work off the clone?

Reply:
I like having the config as immutable all the time because it reinforces the fact that something extra needs to be done to update it.